hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Associator.java

```diff
@@ -65,7 +65,7 @@ public Relationship associationWith(Path path) throws RelationNotFoundException
     }
 
     protected void checkPathLegal(Path targetPath) {
-        if (!context.entityClass.equals(targetPath.getSegment().getElementType())) {
+        if (!context.entityClass.getSimpleName().equals(targetPath.getSegment().getElementType().getSimpleName())) {
```

Comment:
Not against it as long as there is a way of converting an entity **type** to a segment type.